modules/contact/src/ContactMaster.C

```diff
@@ -417,6 +417,9 @@ contactFormulation(const std::string & the_name)
   else if ("augmented_lagrange" == name)
     formulation = CF_AUGMENTED_LAGRANGE;
 
+  else if ("tangential_penalty" == name)
```

Comment:
Also, it kind of looks like you are coding up something that should be handled by a MooseEnum...